library/src/hcc_detail/hipsolver_compat.cpp

```diff
@@ -0,0 +1,139 @@
+/* ************************************************************************
+ * Copyright 2020-2021 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+
+#include "hipsolver.h"
+
+extern "C" {
+
+// gesvd
+hipsolverStatus_t hipsolverDnSgesvd_bufferSize(hipsolverDnHandle_t handle, int m, int n, int* lwork)
+{
+    return hipsolverSgesvd_bufferSize(handle, 'N', 'N', 0, 0, lwork);
+}
+
+hipsolverStatus_t hipsolverDnDgesvd_bufferSize(hipsolverDnHandle_t handle, int m, int n, int* lwork)
+{
+    return hipsolverDgesvd_bufferSize(handle, 'N', 'N', 0, 0, lwork);
+}
+
+hipsolverStatus_t hipsolverDnCgesvd_bufferSize(hipsolverDnHandle_t handle, int m, int n, int* lwork)
+{
+    return hipsolverCgesvd_bufferSize(handle, 'N', 'N', 0, 0, lwork);
+}
+
+hipsolverStatus_t hipsolverDnZgesvd_bufferSize(hipsolverDnHandle_t handle, int m, int n, int* lwork)
+{
+    return hipsolverZgesvd_bufferSize(handle, 'N', 'N', 0, 0, lwork);
+}
+
+hipsolverStatus_t hipsolverDnSgesvd(hipsolverDnHandle_t handle,
+                                    signed char         jobu,
+                                    signed char         jobv,
+                                    int                 m,
+                                    int                 n,
+                                    float*              A,
+                                    int                 lda,
+                                    float*              S,
+                                    float*              U,
+                                    int                 ldu,
+                                    float*              V,
+                                    int                 ldv,
+                                    float*              work,
+                                    int                 lwork,
+                                    float*              rwork,
+                                    int*                devInfo)
+{
+    return hipsolverSgesvd(
+        handle, jobu, jobv, m, n, A, lda, S, U, ldu, V, ldv, NULL, 0, rwork, devInfo);
```

Comment:
Once the bufferSize helper is corrected, we should be able to pass `work` and `lwork` to `hipsolverXgesvd`, as we do on the Nvidia side; so, this implementation should go into the common file.